src/ireiat/data_pipeline/assets/tap/network.py

```diff
@@ -34,5 +42,74 @@ def tap_network_dataframe(
     tap_network = tap_network.sort_values(["tail", "head"])
 
     assert tap_network["speed"].min() > 0
+    context.log.info(f"TAP highway network dataframe created with {len(tap_network)} edges.")
+    publish_metadata(context, tap_network)
+    return tap_network
+
+
+@dagster.asset(
+    io_manager_key="custom_io_manager",
+    metadata={
+        "format": "parquet",
+        "write_kwargs": dagster.MetadataValue.json({"index": False}),
+        **INTERMEDIATE_DIRECTORY_ARGS,
+    },
+)
+def tap_rail_network_dataframe(
+    context: dagster.AssetExecutionContext, rail_network_dataframe: pd.DataFrame
+) -> pd.DataFrame:
+    """Entire network to represent the TAP, complete with capacity and cost information"""
+    # fill out other fields needed for the TAP
+    tap_network = rail_network_dataframe
+
+    if "speed" not in tap_network.columns:
+        # If the 'speed' column doesn't exist
+        tap_network["speed"] = RAIL_DEFAULT_MPH_SPEED
+    else:
+        # Fill missing values with the default value
+        tap_network["speed"] = tap_network["speed"].fillna(RAIL_DEFAULT_MPH_SPEED)
+
+    tap_network["fft"] = tap_network["length"] / tap_network["speed"]
+    tap_network["beta"] = RAIL_BETA
+    tap_network["alpha"] = RAIL_ALPHA
+    tap_network["capacity"] = RAIL_CAPACITY_TONS
```

Comment:
might be worth just adding a TODO to say: deal with actual rail capacity based on track count